org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/debugmodel/DSPDebugTarget.java

```diff
@@ -632,8 +587,24 @@ public void exited(ExitedEventArguments args) {
 
 	@Override
 	public void thread(ThreadEventArguments args) {
-		refreshThreads.set(true);
-		fireChangeEvent(DebugEvent.CONTENT);
+		triggerUpdateThreads();
```

Comment:
Note that in case where multiple threads are started in quick succession this will cause multiple threads() calls to be made in quick succession.